src/likelihoodratiotest.jl

```diff
@@ -215,3 +255,49 @@ function StatsModels.isnested(m1::MixedModel, m2::MixedModel; atol::Real=0.0)
 
     true
 end
+
+
+"""
+    isnested(m1::LinearModel, m2::LinearMixedModel; atol::Real=0.0)
+    isnested(m1::TableRegressioLinearModel{LinearModel}, m2::LinearMixedModel; atol::Real=0.0)
+    isnested(m1::GeneralizedLinearModel, m2::GeneralizedLinearMixedModel; atol::Real=0.0)
+    isnested(m1::TableRegressioLinearModel{GeneralizedLinearModel}, m2::GeneralizedLinearMixedModel; atol::Real=0.0)
+
+Indicate whether model `m1` is nested in model `m2`, i.e. whether
+`m1` can be obtained by constraining some parameters in `m2`.
+Both models must have been fitted on the same data. The mixed model must be
+fitted with maximum likelihood (i.e. **not** REML).
+"""
+function StatsModels.isnested(m1::TableRegressionModel{<:Union{LinearModel, GeneralizedLinearModel}},
+                              m2::MixedModel; atol::Real=0.0)
+    StatsModels.isnested(m1.model, m2) || return false
+
+    # check that the nested fixef are a subset of the outer
+    all(in.(coefnames(m1),  Ref(coefnames(m2)))) || return false
+
+    return true
+end
+
+# GLM isn't nested with in LMM and LM isn't nested within GLMM
+StatsModels.isnested(m1::Union{LinearModel, GeneralizedLinearModel}, m2::MixedModel; atol::Real=0.0) = false
+
+function StatsModels.isnested(m1::LinearModel, m2::LinearMixedModel; atol::Real=0.0)
+    nobs(m1) == nobs(m2) || return false
+
+    !m2.optsum.REML ||
+        throw(ArgumentError("REML-fitted models cannot be compared to linear models"))
+
+    return true
+end
+
+function StatsModels.isnested(m1::GeneralizedLinearModel, m2::GeneralizedLinearMixedModel; atol::Real=0.0)
+    nobs(m1) == nobs(m2) || return false
+
+    Distribution(m1) == Distribution(m2) ||
+        throw(ArgumentError("Models must be fit to the same distribution"))
+
+    Link(m1) == Link(m2) ||
+        throw(ArgumentError("Models must have the same link function"))
+
+    return true
+end
```

Comment:
For these could you check somehow the width of the design matrix (just the fixed part for the mixed models)

Reply:
Hmmm.... yeah. That sounds like a basic safety check.